### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by silkeh**
This PR adds configuration for [golangci-lint](https://golangci-lint.run), and enables/fixes all low-hanging fruit.

### CODE REVIEW

File: builder/manager.go [hunk 1]
```diff
@@ -417,8 +437,9 @@ func (m *Manager) Index(dir string) error {
 	// Now set our options according to the config
 	m.overlay.EnableTmpfs = m.Config.EnableTmpfs
 	m.overlay.TmpfsSize = m.Config.TmpfsSize
-	if ValidMemSize(m.overlay.TmpfsSize) != true && m.overlay.EnableTmpfs == true {
-		log.Fatalf("Invalid memory size specified: %s\n", m.overlay.TmpfsSize)
+
+	if !ValidMemSize(m.overlay.TmpfsSize) && m.overlay.EnableTmpfs {
+		log.Panicf("Invalid memory size specified: %s\n", m.overlay.TmpfsSize)
```
Comment (EbonJaeger) [inline builder/manager.go:442]: Why the change from `Fatalf` to `Panicf`?

Comment (silkeh) [reply builder/manager.go:442]: Fatal calls `os.Exit`, causing any `defer` calls to not be executed. Using `panic` still fatally exits, but also executes the `defer`s.

**Review by EbonJaeger — APPROVED** [review]
LGTM, thanks! :)
